core/feconf.py

```diff
@@ -24,11 +24,31 @@
 
 from core.constants import constants
 
-from typing import Dict, List, Union
+from typing import Dict, List
 from typing_extensions import TypedDict
 
-CommandType = (
-    Dict[str, Union[str, List[str], Dict[str, Union[str, List[str]]]]])
+
+# This TypedDict is used only to create the TypedDict below, because we want
+# to mark some attributes as optional.
+class _ValidCmdAttributeSpecsAttributeDictNotTotal(TypedDict, total=False):
+    """Dictionary representing a non-required part
+    of valid commands specs."""
+
+    allowed_values: Dict[str, List[str]]
+    deprecated_values: Dict[str, List[str]]
+
+
+class ValidCmdAttributeSpecsAttributeDicts(
```

Comment:
Actually... I can't really parse this name, sorry. Would it be possible to express this concept in a simpler way?

In particular I think it's the repeated "attribute" in "attribute specs attribute dicts" that is throwing me off. Also, I can understand something like "ValidCmdDict" but I don't know what an "attribute dict" is since all dicts will have attributes (assuming we're talking about the keys).